rules/rollup.go

```diff
@@ -94,10 +78,26 @@ func newRollupTarget(target *schema.RollupTarget) (rollupTarget, error) {
 	}, nil
 }
 
+func newRollupTargetFromView(rtv models.RollupTargetView) rollupTarget {
+	return rollupTarget{
+		Name:     []byte(rtv.Name),
+		Tags:     bytesArrayFromStringArray(rtv.Tags),
+		Policies: rtv.Policies,
+	}
+}
+
+func (t rollupTarget) rollupTargetView() models.RollupTargetView {
+	return models.RollupTargetView{
+		Name:     string(t.Name),
+		Tags:     stringArrayFromBytesArray(t.Tags),
+		Policies: t.Policies,
+	}
+}
+
 // SameTransform returns whether two rollup targets have the same transformation.
-func (t *rollupTarget) SameTransform(other rollupTarget) bool {
-	tView := newRollupTargetView(*t)
-	otherView := newRollupTargetView(other)
+func (t rollupTarget) SameTransform(other rollupTarget) bool {
```

Comment:
We need to be careful w.r.t. performance here as this is called on the hot patch of rule matching in the collectors. By converting them to `rollupTargetView`, we are making a first copy of the underlying names and tags, and then by calling `SameTransform` on the view, we are making a second copy of the underlying tags, which is inefficient.

Can you perhaps make this unexported leave the implementation as is for now? I'll take a pass at the rule matching logic but IIRC we may not need to call `sameTrasnform` on `rollupTarget` during rule matching as it should've already been validated at rule creation time, in which case we can simply remove this function.

FYI, the original implementation is
```
func (t *RollupTarget) sameTransform(other RollupTarget) bool {
	if !bytes.Equal(t.Name, other.Name) {
		return false
	}
	if len(t.Tags) != len(other.Tags) {
		return false
	}
	clonedTags := stringArrayFromBytesArray(t.Tags)
	sort.Strings(clonedTags)
	otherClonedTags := stringArrayFromBytesArray(other.Tags)
	sort.Strings(otherClonedTags)
	for i := 0; i < len(clonedTags); i++ {
		if clonedTags[i] != otherClonedTags[i] {
			return false
		}
	}
	return true
}
```